app/src/main/java/com/andela/helpmebuy/adapters/ReceivedRequestAdapter.java

```diff
@@ -42,30 +42,29 @@ public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     public void onBindViewHolder(final CustomViewHolder holder, final int position) {
         DateManager dateManager = new DateManager();
         PurchaseRequest request = purchaseRequests.get(position);
-        holder.sender.setText(request.getSendersFullName());
         String items = getItemsDescription(request);
+        String date = dateManager.formatTime(request.getDateMillis());
+        holder.sender.setText(request.getSendersFullName());
         holder.description.setText(items);
-        holder.date.setText(dateManager.formatTime(request.getDateMillis()));
+        holder.date.setText(date);
     }
 
     public String getItemsDescription(PurchaseRequest request){
         String itemName = "";
         ArrayList<PurchaseItem> items = request.getPurchaseList();
-        if(items.size() == 1) {
-            return itemName + items.get(0).getItemName();
+        if(items.size() > 0) {
```

Comment:
Technically, that means if the number of items is 2, condition 1 would be satisfied and also condition 2 and both would run because it is `> 0` and also `>=2`. But we don't want it to go inside the first condition, we want it in the second.